### PR TITLE
port: 6 to 7 20221111

### PR DESCRIPTION
**Pull request by nkoenig**
# ➡️ Forward port

Port `ign-gazebo6` to `gz-sim7`.

Most of the changes are ign->gz naming updates. Here is a list of files that contain changes other changes:

1. `Changelog.md`
2.  `examples/worlds/tunnel.sdf`
3.  `src/systems/rf_comms/RFComms.cc`
4. `test/integration/hydrodynamics.cc`
5. `test/worlds/hydrodynamics.sdf`

@azeey and/or @scpeters can you take a look at the Physics system. I combined the merge, which I think is making the tests upset.

@caguero , can you give the thruster changes a look?

Branch comparison: [https://github.com/gazebosim/<REPO>/compare/<TO_BRANCH>...<FROM_BRANCH>](https://github.com/gazebosim/gz-sim/compare/gz-sim7...ign-gazebo6)

**Note to maintainers**: Remember to **Merge** with commit (not squash-merge or rebase)

### CODE REVIEW

File: Changelog.md [hunk 1]
```diff
@@ -656,1880 +721,1880 @@
 1. Set simulation time to Rendering
     * [Pull request #1415](https://github.com/gazebosim/gz-sim/pull/1415)
 
-### Gazebo Sim 6.9.0 (2022-04-14)
+### Ignition Gazebo 6.9.0 (2022-04-14)
```
Comment (scpeters) [inline Changelog.md:724]: these gazebo -> ignition changes should be reverted from the changelog

File: src/systems/physics/Physics.cc [hunk 2]
```diff
@@ -161,38 +161,39 @@ class gz::sim::systems::PhysicsPrivate
...
-          gz::physics::GetLinkFromModel,
-          gz::physics::GetShapeFromLink
+  public: struct MinimumFeatureList : physics::FeatureList<
+          physics::FindFreeGroupFeature,
+          physics::SetFreeGroupWorldPose,
+          physics::FreeGroupFrameSemantics,
+          physics::LinkFrameSemantics,
+          physics::ForwardStep,
+          physics::RemoveModelFromWorld,
+          physics::sdf::ConstructSdfLink,
```
Comment (scpeters) [inline src/systems/physics/Physics.cc:171]: revert this `ConstructSdfLink` line

File: src/systems/physics/Physics.cc [hunk 3]
```diff
@@ -458,18 +459,19 @@ class gz::sim::systems::PhysicsPrivate
...
             MinimumFeatureList,
-            gz::physics::GetJointFromModel,
-            gz::physics::GetBasicJointProperties,
-            gz::physics::GetBasicJointState,
-            gz::physics::SetBasicJointState>{};
+            physics::GetJointFromModel,
+            physics::GetBasicJointProperties,
+            physics::GetBasicJointState,
+            physics::SetBasicJointState,
+            physics::sdf::ConstructSdfJoint>{};
```
Comment (scpeters) [inline src/systems/physics/Physics.cc:474]: revert this `ConstructSdfJoint` line

File: src/gui/plugins/scene_manager/GzSceneManager.cc [hunk 4]
```diff
@@ -132,6 +132,12 @@ void GzSceneManager::Update(const UpdateInfo &_info,
     this->dataPtr->renderEnginePluginPathsInit = true;
   }
 
+  if (!this->dataPtr->renderEnginePluginPathsInit)
+  {
+    this->dataPtr->renderUtil.InitRenderEnginePluginPaths();
+    this->dataPtr->renderEnginePluginPathsInit = true;
+  }
```
Comment (scpeters) [inline src/gui/plugins/scene_manager/GzSceneManager.cc:139]: this block is a duplicate of the one right above. Is this intentional?

File: Changelog.md (hunk 1 above)
Comment (nkoenig) [reply Changelog.md]: [ffe2da0](https://github.com/gazebosim/gz-sim/pull/1794/commits/ffe2da087fb41925a0fab13ee063774c1c93cae0)

File: src/systems/physics/Physics.cc (hunk 2 above)
Comment (nkoenig) [reply src/systems/physics/Physics.cc]: [ffe2da0](https://github.com/gazebosim/gz-sim/pull/1794/commits/ffe2da087fb41925a0fab13ee063774c1c93cae0)

File: src/gui/plugins/scene_manager/GzSceneManager.cc (hunk 4 above)
Comment (nkoenig) [reply src/gui/plugins/scene_manager/GzSceneManager.cc]: [ffe2da0](https://github.com/gazebosim/gz-sim/pull/1794/commits/ffe2da087fb41925a0fab13ee063774c1c93cae0)

**Review by caguero — COMMENTED** [review]
I tested my code that uses thrusters and seems to work. I just noticed a warning that didn't see in the previous branch (`6-to-7-20221021`):

```
[ruby $(which gz) sim-1] [Err] [SDFFeatures.cc:320] Mesh construction from an SDF has not been implemented yet for dartsim.
[ruby $(which gz) sim-1] [Err] [SDFFeatures.cc:787] The geometry element of collision [collision] couldn't be created
```

**Comment by caguero** [comment]
> I tested my code that uses thrusters and seems to work. I just noticed a warning that didn't see in the previous branch (`6-to-7-20221021`):
> 
> ```
> [ruby $(which gz) sim-1] [Err] [SDFFeatures.cc:320] Mesh construction from an SDF has not been implemented yet for dartsim.
> [ruby $(which gz) sim-1] [Err] [SDFFeatures.cc:787] The geometry element of collision [collision] couldn't be created
> ```

This is coming from `gz-physics`, so probably unrelated due to a `vcs pull`.

**Comment by scpeters** [comment]
I just restarted the windows CI job after wiping the workspace

I think the commit history would be a little tidier if all the commits responding to PR feedback were squashed into https://github.com/gazebosim/gz-sim/pull/1794/commits/878625dfa111a45ad4eb41bd0d6ca5b6bfd1101e

**Comment by mjcarroll** [comment]
> I think the commit history would be a little tidier if all the commits responding to PR feedback were squashed

Agree, otherwise looks good to me.

**Comment by nkoenig** [comment]
I squashed the commits down into 78b66994bbef46d0d03917e935effe8886107343, but was unable to figure out how to squash them into 878625dfa111a45ad4eb41bd0d6ca5b6bfd1101e. I was using the rebase command.

**Review by caguero — APPROVED** [review]

**Comment by nkoenig** [comment]
> The physics system looks good. I only saw namespace changes.
> 
> To squash into [878625d](https://github.com/gazebosim/gz-sim/commit/878625dfa111a45ad4eb41bd0d6ca5b6bfd1101e), I think you can do
> 
> ```
> git reset --soft 878625d
> git commit --amend
> ```

Thanks.

**Review by mjcarroll — APPROVED** [review]

**Comment by nkoenig** [comment]
The failing tests are all related to fuelserver timeouts.

**Pull request merged**